File: market/test/unit/java/ru/yandex/market/logistics/utilizer/service/system/SystemPropertyServiceUnitTest.java
package ru.yandex.market.logistics.utilizer.service.system;

import java.util.Optional;
import java.util.Set;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.mockito.Mockito;

import ru.yandex.market.logistics.utilizer.base.SoftAssertionSupport;
import ru.yandex.market.logistics.utilizer.domain.entity.SystemProperty;
import ru.yandex.market.logistics.utilizer.repo.SystemPropertyJpaRepository;
import ru.yandex.market.logistics.utilizer.service.system.keys.SystemPropertySetIntegerKey;
import ru.yandex.market.logistics.utilizer.service.system.keys.SystemPropertySetLongKey;

public class SystemPropertyServiceUnitTest extends SoftAssertionSupport {

    private SystemPropertyService systemPropertyService;
    private SystemPropertyJpaRepository systemPropertyRepository;

    @BeforeEach
    public void init() {
        systemPropertyRepository = Mockito.mock(SystemPropertyJpaRepository.class);
        systemPropertyService = new SystemPropertyService(systemPropertyRepository);
        Mockito.when(systemPropertyRepository
                .findByName(SystemPropertySetLongKey.VENDORS_TO_FINALIZE_CYCLE_FOR.name()))
                .thenReturn(Optional.of(createSystemProperty(" 123, 456   ")));
        Mockito.when(systemPropertyRepository
                .findByName(SystemPropertySetLongKey.VENDORS_TO_CREATE_TRANSFER_FOR.name()))
                .thenReturn(Optional.of(createSystemProperty(" ")));
    }

    @Test
    public void systemPropertySetLongForNotEmptyString() {
        Set<Long> actual = systemPropertyService.getProperty(SystemPropertySetLongKey.VENDORS_TO_FINALIZE_CYCLE_FOR);
        softly.assertThat(actual).containsExactlyInAnyOrder(123L, 456L);
    }

    @Test
    public void systemPropertySetLongForEmptyString() {
        Set<Long> actual = systemPropertyService.getProperty(SystemPropertySetLongKey.VENDORS_TO_CREATE_TRANSFER_FOR);
        softly.assertThat(actual).isEmpty();
    }

    @Test
    public void systemPropertySetIntegerForNotEmptyString() {
        Mockito.when(systemPropertyRepository
                .findByName(SystemPropertySetIntegerKey.WAREHOUSES_FOR_UTILIZATION.name()))
                .thenReturn(Optional.of(createSystemProperty(" 123, 456   ")));
        Set<Integer> actual = systemPropertyService.getProperty(SystemPropertySetIntegerKey.WAREHOUSES_FOR_UTILIZATION);
        softly.assertThat(actual).containsExactlyInAnyOrder(123, 456);
    }

    @Test
    public void systemPropertySetIntegerForEmptyString() {
        Mockito.when(systemPropertyRepository
                .findByName(SystemPropertySetIntegerKey.WAREHOUSES_FOR_UTILIZATION.name()))
                .thenReturn(Optional.of(createSystemProperty("")));
        Set<Integer> actual = systemPropertyService.getProperty(SystemPropertySetIntegerKey.WAREHOUSES_FOR_UTILIZATION);
        softly.assertThat(actual).isEmpty();
    }

    private SystemProperty createSystemProperty(String value) {
        SystemProperty systemProperty = new SystemProperty();
        systemProperty.setValue(value);
        return systemProperty;
    }
}

File: market/test/unit/kotlin/service/system/SystemPropertyServiceUnitTest.kt
package ru.yandex.market.logistics.calendaring.service.system

import org.junit.jupiter.api.Test
import org.mockito.Mockito
import ru.yandex.market.logistics.calendaring.base.SoftAssertionSupport
import ru.yandex.market.logistics.calendaring.model.entity.SystemProperty
import ru.yandex.market.logistics.calendaring.repository.SystemPropertyJpaRepository
import ru.yandex.market.logistics.calendaring.service.system.keys.SystemPropertySetIntegerKey
import ru.yandex.market.logistics.calendaring.service.system.keys.SystemPropertySetLongKey


class SystemPropertyServiceUnitTest : SoftAssertionSupport() {

    private var systemPropertyRepository: SystemPropertyJpaRepository =
        Mockito.mock(SystemPropertyJpaRepository::class.java)
    private var systemPropertyService: SystemPropertyService = SystemPropertyService(systemPropertyRepository)


    @Test
    fun systemPropertySetLongForNotEmptyString() {
        Mockito.`when`(
            systemPropertyRepository
                .findByName(SystemPropertySetLongKey.FEATURE_GATE_INDIVIDUAL_SCHEDULE_ENABLED_FOR_WAREHOUSES.name)
        )
            .thenReturn(createSystemProperty(" 123, 456   "))

        val actual: Set<Long> =
            systemPropertyService.getProperty(SystemPropertySetLongKey.FEATURE_GATE_INDIVIDUAL_SCHEDULE_ENABLED_FOR_WAREHOUSES)!!
        softly.assertThat(actual).containsExactlyInAnyOrder(123L, 456L)
    }

    @Test
    fun systemPropertySetLongForEmptyString() {
        Mockito.`when`(
            systemPropertyRepository
                .findByName(SystemPropertySetLongKey.FEATURE_GATE_INDIVIDUAL_SCHEDULE_ENABLED_FOR_WAREHOUSES.name)
        )
            .thenReturn(createSystemProperty(" "))

        val actual: Set<Long> =
            systemPropertyService.getProperty(SystemPropertySetLongKey.FEATURE_GATE_INDIVIDUAL_SCHEDULE_ENABLED_FOR_WAREHOUSES)!!
        softly.assertThat(actual).isEmpty()
    }

    @Test
    fun systemPropertySetIntegerForNotEmptyString() {
        Mockito.`when`(
            systemPropertyRepository
                .findByName(SystemPropertySetIntegerKey.EXAMPLE.name)
        )
            .thenReturn(createSystemProperty(" 123, 456   "))
        val actual: Set<Int> = systemPropertyService.getProperty(SystemPropertySetIntegerKey.EXAMPLE)
        softly.assertThat(actual).containsExactlyInAnyOrder(123, 456)
    }

    @Test
    fun systemPropertySetIntegerForEmptyString() {
        Mockito.`when`(
            systemPropertyRepository
                .findByName(SystemPropertySetIntegerKey.EXAMPLE.name)
        )
            .thenReturn(createSystemProperty(""))
        val actual: Set<Int> = systemPropertyService.getProperty(SystemPropertySetIntegerKey.EXAMPLE)
        softly.assertThat(actual).isEmpty()
    }

    private fun createSystemProperty(value: String): SystemProperty {
        val systemProperty = SystemProperty()
        systemProperty.value = value
        return systemProperty
    }
}
